src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java

```diff
@@ -198,6 +202,19 @@ private static String getFilePathSuffix() {
             return descriptors;
         }
 
+        @Override public UninstantiatedDescribable customUninstantiate(UninstantiatedDescribable ud) {
+            Object scm = ud.getArguments().get("scm");
+            if (scm instanceof UninstantiatedDescribable) {
+                UninstantiatedDescribable scmUd = (UninstantiatedDescribable) scm;
+                Map<String, Object> scmArguments = new HashMap<>(scmUd.getArguments());
+                scmArguments.remove("id");
```

Comment:
So IIUC, the problem is that we don't want this ID to be part of the model, since it should be unique to each instance?

Reply:
Right. `SCMSource` does a weird trick that does not work like your typical `Describable`: it generates a random `id` each time you instantiate it, but also lets the ID be customized.